sockeye/inference.py

```diff
@@ -514,30 +538,37 @@ def get_max_output_length(input_length: int):
 
 
 Tokens = List[str]
-TranslatorInput = NamedTuple('TranslatorInput', [
-    ('id', int),
-    ('sentence', str),
-    ('tokens', Tokens),
-])
-"""
-Required input for Translator.
-
-:param id: Sentence id.
-:param sentence: Input sentence.
-:param tokens: List of input tokens.
-"""
 
-InputChunk = NamedTuple("InputChunk",
-                        [("id", int),
-                         ("chunk_id", int),
-                         ("tokens", Tokens)])
-"""
-A chunk of a TranslatorInput.
+class TranslatorInput:
+    """
+    Required input for Translator.
 
-:param id: Sentence id.
-:param chunk_id: The id of the chunk.
-:param tokens: List of input tokens.
-"""
+    :param id: Sentence id.
+    :param sentence: Input sentence.
+    :param tokens: List of input tokens.
+    :param factors: List of zero or more input factors.
+    """
+    def __init__(self, id, sentence, tokens, chunk_id = 0, factors = []):
```

Comment:
typing information missing.
`id` is a reserved word, maybe we can rename it?

Reply:
this is used elsewhere as well (e.g., TranslatedChunk), does this matter if it's just a function def? If you change it, we also have to change it in calls where the ID is passed

Reply:
That'd be easy with an IDE :) 
Agreed that it is not crucial in this case. Your decision whether you want to change it.